pkg/mads/server/mads.go

```diff
@@ -0,0 +1,257 @@
+package server
+
+import (
+	"context"
+	"strconv"
+	"sync/atomic"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+
+	"github.com/golang/protobuf/proto"
+	any "github.com/golang/protobuf/ptypes/any"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	kuma_observability "github.com/Kong/kuma/api/observability/v1alpha1"
+	"github.com/Kong/kuma/pkg/mads"
+
+	envoy "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	envoy_core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	envoy_cache "github.com/envoyproxy/go-control-plane/pkg/cache"
+	envoy_server "github.com/envoyproxy/go-control-plane/pkg/server"
+)
+
+type Server interface {
```

Comment:
I see this code is very similar to SDS. Can we somehow avoid duplication? Of course, if it doesn't make sense then don't do it. Then please at least provide link on which code this implementation is based on.

Reply:
I don't want to invest time into this generalisation right now